server/src/controllers/Events/resolver.ts

```diff
@@ -854,17 +854,34 @@ ${unsubscribeOptions}`,
     });
     await deleteEventReminders(id);
 
+    const chapterURL = `${process.env.CLIENT_LOCATION}/chapters/${event.chapter.id}`;
+    const eventURL = `${process.env.CLIENT_LOCATION}/events/${event.id}?emaillink=true`;
     const notCanceledRsvps = event.event_users;
 
     if (notCanceledRsvps.length) {
       const emailList = notCanceledRsvps.map(({ user }) => user.email);
-      const subject = `Event ${event.name} canceled`;
-      const body = `The event ${event.name} was canceled`;
+      const subject = `Event ${event.name} is canceled`;
+
+      // ToDo: change "subscribed to event" to "join event" when join and leave function is on
+
+      const cancelEventEmail = `The Upcoming Event ${event.name} is canceled.<br />
+      <br />
+      View Upcoming Events for ${event.chapter.name}: <a href='${chapterURL}'>${event.chapter.name} chapter</a>.<br />
+      ----------------------------<br />
+      <br />
+      - Stop receiving event notifications for ${event.name}. You can do it here: <a href="${eventURL}">${eventURL}</a>.<br />
+      - More about ${event.chapter.name} or to unfollow this chapter: <a href="${chapterURL}">${chapterURL}</a>.<br />
+      <br />
+      ----------------------------<br />
+      You received this email because you Subscribed to ${event.name} Event.<br />
+      <br />
+      See the options above to change your notifications.
```

Comment:
```suggestion
      const cancelEventEmail = `The upcoming event ${event.name} has been canceled.<br />
      <br />
      View upcoming events for ${event.chapter.name}: <a href='${chapterURL}'>${event.chapter.name} chapter</a>.<br />
      ----------------------------<br />
```
I think this flows better. I deleted the notifications text because it should be possible to DRY it out.  

We have two types of emails

- Ones sent in response to a user action
- Ones sent because a user is subscribed

All emails should have the same unsubscribe text

```
To manage notifications about this event, go to <a href="${eventURL}">${eventURL}</a>.<br />
To manage notifications from this chapter about new events, go to <a href="${chapterURL}">${chapterURL}</a>.<br />
```

Emails that are sent because the user is subscribed to an event should explain that.  Similarly emails that are sent because of a chapter subscription should explain that.

Finally emails that are sent in response to a user action need no explanation.

@Sboonny could you handle this in this PR?  I think makes sense to handle them all together, so we can make sure they're consistent.

Reply:
I have tried to DRY it using `unsubscribeOption`, but the resolver didn't use `user.id`. 

So before I add them, do we need `user.id` in cancel events email? although it doesn't make sense to cancel cancel events, people may want to cancel their chapter membership.

So do we add it? or just link it?

Reply:
It seems simplest to always provide both of the unsubscribe links. We don't need to customise them for the users current subscription state.

Unfollowing cancelled events should mostly be pointless, but conceivably events could be uncancelled. It's safer to let users unsubscribe and respect their wishes.

Reply:
Sure, random question 😅, the https://github.com/freeCodeCamp/chapter/commit/06209830d3c450a4d4096bfdf2d3cbf61cb47e11 should have failed cypress because there were no `user.id`, but it succeeded should. should we worry about that?

Reply:
Given that the tests didn't run, I'm not worried!

Reply:
sorry for the confusion, the function meant to run without user.id, which is interesting 🤔